FunctionalTableData/TableSection.swift

```diff
@@ -164,6 +164,20 @@ extension Array where Element: TableSectionType {
 
 extension TableSection: Equatable {
 	public static func ==(lhs: TableSection, rhs: TableSection) -> Bool {
-		return lhs.key == rhs.key
+		return lhs.key == rhs.key &&
+			(lhs.header == nil && rhs.header == nil || lhs.header?.isEqual(rhs.header) ?? false) &&
```

Comment:
Why are you checking for the nil? Wouldn't isEqual do this too?

Reply:
`TableHeaderFooterConfigType` is not equatable but has a `isEqual` method defined for it. This means the `isEqual` method call has to handle nil here.

Reply:
What I mean is why not just use this

`lhs.header?.isEqual(rhs.header)`. Wouldn't isEqual return false if one of them is `nil`?

Reply:
If one of them is nil and the other isn't then `isEqual` should return false, but if both are nil then `isEqual` should return true. That line of code you suggest doesn't handle that case.